lib/assets/javascripts/twine.coffee

```diff
@@ -69,13 +70,29 @@
       element = findOrCreateElementForNode(node)
       element.indexes = indexes
 
-    for type, binding of Twine.bindingTypes when definition = Twine.getAttribute(node, type)
-      element = findOrCreateElementForNode(node)
+    bindingConstructors = null
+    for attribute in node.attributes
+      type = attribute.name
+      type = type.slice(5) if type.slice(0, 5) == 'data-'
+      definition = attribute.value
+
+      constructor = Twine.bindingTypes[type]
+      continue unless constructor
+
+      bindingConstructors ?= []
+      if type == 'bind'
```

Comment:
I made a quick perf test to compare some different ways. Checking each char like @justin-gomes suggested is fastest:
```javascript
indexOfPerf = function(t) {
  var start = new Date;
  for (var i = 0; i < t; i++) {
    "data-bind-event".indexOf("data-") == 0;
    "class".indexOf("data-") == 0;
  }
  console.log("indexOf took", new Date - start, "ms to run", t, "times");
}

slicePerf = function(t) {
  var start = new Date;
  for (var i = 0; i < t; i++) {
    "data-bind-event".slice(5) == "data-";
    "class".slice(5) == "data-";
  }
  console.log("slice took", new Date - start, "ms to run", t, "times");
}

charAtSlicePerf = function(t) {
  var start = new Date;
  for (var i = 0; i < t; i++) {
    "data-bind-event"[4] == "-" && "data-bind-event".slice(5) == "data-";
    "class"[4] == "-" && "class".slice(5) == "data-";
  }
  console.log("char and slice took", new Date - start, "ms to run", t, "times");
}

startsWithPerf = function(t) {
  var start = new Date;
  for (var i = 0; i < t; i++) {
    "data-bind-event".startsWith("data-");
    "class".startsWith("data-");
  }
  console.log("startsWith took", new Date - start, "ms to run", t, "times");
}

eachCharPerf = function(t) {
  var start = new Date;
  for (var i = 0; i < t; i++) {
    "data-bind-event"[0] == "d" && "data-bind-event"[1] == "a" && "data-bind-event"[2] == "t" && "data-bind-event"[3] == "a" && "data-bind-event"[4] == "-";
    "class"[0] == "d" && "class"[1] == "a" && "class"[2] == "t" && "class"[3] == "a" && "class"[4] == "-";
  }
  console.log("each char took", new Date - start, "ms to run", t, "times");
}

var times = 10000000;
indexOfPerf(times);
slicePerf(times);
charAtSlicePerf(times);
startsWithPerf(times); // not supported in IE
eachCharPerf(times);
```
```
indexOf took 1248 ms to run 10000000 times
VM4517:16 slice took 588 ms to run 10000000 times
VM4517:25 char and slice took 370 ms to run 10000000 times
VM4517:34 startsWith took 742 ms to run 10000000 times
VM4517:43 each char took 5 ms to run 10000000 times
```